src/renderers/native/NativeMethodsMixin.js

```diff
@@ -150,9 +150,9 @@ var NativeMethodsMixin = {
     );
 
     UIManager.updateView(
-      findNodeHandle(this),
+      (findNodeHandle(this) : any),
       this.viewConfig.uiViewClassName,
-      updatePayload
+      (updatePayload : any)
```

Comment:
Hmm. I didn't modify this code other than to add the type-cast. But you're right. It looks fishy.

`ReactNativeAttributePayload.create` may return `null` if no props have changed. So that return type is correct.

The Java implementation of `updateView` doesn't specify a `@Nullable` annotation for the incoming payload param, _but_ the underlying `UIImplementation` class [does check for null]https://phabricator.intern.facebook.com/diffusion/FBS/browse/master/fbandroid/java/com/facebook/catalyst/js/react-native-github/ReactAndroid/src/main/java/com/facebook/react/uimanager/UIImplementation.java;9de7fef3c4b2d1a90f8b907b0cde843f4a221493$212) before doing anything with them.

I think the right thing to do here though, regardless, is to check for non-null before calling. I'll do that.